test/ably/restauth_test.py

```diff
@@ -196,14 +196,30 @@ def test_if_authorize_changes_auth_mechanism_to_token(self):
                          msg="Authorise should change the Auth method")
 
     # RSA10a
+    @dont_vary_protocol
     def test_authorize_always_creates_new_token(self):
-        token = self.ably.auth.authorize()
-        new_token = self.ably.auth.authorize()
+        # Token with short ttl
+        ttl = TokenDetails.TOKEN_EXPIRY_BUFFER + 1000
+        token_details = self.ably.auth.request_token(token_params={'ttl': ttl})
+
+        # Client with no means to renew
+        ably = AblyRest(token_details=token_details,
+                        rest_host=test_vars["host"],
+                        port=test_vars["port"],
+                        tls_port=test_vars["tls_port"],
+                        tls=test_vars["tls"])
+
+        # First request passes
+        ably.request('GET', '/time')
+        time.sleep(1)
 
-        self.assertGreater(token.expires, time.time()*1000)
-        self.assertIsNot(new_token, token)
+        # Second does not
+        with self.assertRaises(AblyAuthException):
```

Comment:
I am not sure this is correct if it fails?  What is the value of `TOKEN_EXPIRY_BUFFER`?  If it's say 2s, this test is very racy as the realtime system will still respond if the token is valid i.e. there is no buffer factored in and the TTL is literal.